compiler/vm/vmdef.nim

```diff
@@ -152,19 +153,20 @@ type
     vmstateDiff*: seq[(PSym, PNode)] # we remember the "diff" to global state here (feature for IC)
     procToCodePos*: Table[int, int]
 
-  PStackFrame* = ref TStackFrame
-  TStackFrame* {.acyclic.} = object
+  StackFrameIndex* = int
+
+  TStackFrame* = object
     prc*: PSym                 # current prc; proc that is evaluated
     slots*: seq[TFullReg]      # parameters passed to the proc + locals;
                               # parameters come first
-    next*: PStackFrame         # for stacking
+    next*: StackFrameIndex         # for stacking
```

Comment:
`next` could use `Option[StackFrameIndex]` instead of just `StackFrameIndex`. This would have the advantage of the default value meaning "no previous frame" (not the case right now), but would also require more space and make access a bit slower.

Reply:
A sentinel index would work as well?

Reply:
Hmm, the real trick would be extending optional, or at least a new data type, that allows for "in range" sentinel/nil like values.